pkg/obfuscator/domain.go

```diff
@@ -44,23 +47,17 @@ func (d *domainObfuscator) replaceDomains(input string) string {
 				replacement = obfuscatedBaseDomain
 			}
 			output = strings.ReplaceAll(output, m[0], replacement)
-			d.AddReplacement(m[0], replacement)
 		}
 	}
 	return output
 }
 
 func (d *domainObfuscator) obfuscatedDomain(domain string) string {
-	if replacement, ok := d.domainMapping[domain]; ok {
-		return replacement
-	}
-	d.domainCount++
-	replacement := fmt.Sprintf(obfuscatedTemplate, d.domainCount)
-	d.domainMapping[domain] = replacement
+	replacement := d.obfsGenerator.generateReplacement(d.replacementType, domain, d.ReplacementTracker)
```

Comment:
`d.replacementType` shouldn't be needed anymore

Reply:
Addressed